### ISSUE TITLE
Non-lowercase repo names not being found

### ISSUE THREAD

**Issue by DanielEWeeks** [repobee/repobee#1171]
I let my students create their assignment repositories by accepting a GitHub Classroom assignment.

So I'm not using repobee to create the assignment repositories, but rather enabling peer review using repobee after the fact.

When I do this, it names the repositories in the style repo-GitHubID where the GitHubID part may contain a mixture of upper and lower case letters.

For example, if the GitHub ID were `DanielWeeks`  and the GitHub Classroom assignment repository base name were `github-intro`, then GitHub Classroom would create a student-specific assignment with the name `github-intro-DanielWeeks`.

However, if I then do

```
repobee reviews assign -a github-intro --num-reviews 1 --issue review_issue.md
```

it fails while "Fetching teams and repos" with the error message:

```
"[ERROR] NotFoundError: Can't find repos: github-intro-danielweeks"
```

Looks like repobee is expecting lower-case GitHub IDs.

This is using the current version of `repobee`, version 3.8.1.

I see several GitHub Issues where it sounds like the decision was made to lowercase things - I suspect that is now what is causing problems in my particular situation. 

I was previously using version 3.2.3, where this problem did not occur (however, that old version is now running into "You have exceeded a secondary rate limit" errors).  

QUESTION: Where is the lowering of the GitHub ID names (that are read from in the student file) being done?  Can I just alter that line to turn it off?

**Comment by DanielEWeeks** [comment]
If I make the following changes, then repobee appears to work OK on my set of student repositories (although I haven't fully tested it, as I was the only one 'student' who submitted a GitHub issue).

Comment out these lines in platform.py in

/Users/dweeks/.repobee/env/lib/python3.9/site-packages/repobee_plug

```
#   def __post_init__(self):
#       object.__setattr__(self, "members", [m.lower() for m in self.members])
```

```
#   def __post_init__(self):
#       object.__setattr__(
#           self,
#           "author",
#           self.author.lower() if self.author is not None else None,
#       )
```

and in localreps.py

```
    def __post_init__(self):
#       object.__setattr__(self, "members", [m.lower() for m in self.members])
        object.__setattr__(
            self, "name", self.name or "-".join(sorted(self.members))
        )
        _check_name_length(self.name)
```

**Comment by slarse** [comment]
This is related to #1014, #903, #900

You're right that RepoBee assumes lowercased names for everything. There were frequent problems with users entering usernames or task names with the wrong case, causing lookup failures.

As the decision was reached to lowercase stuff for convenience, and I can't walk that decision back without breaking changes, I think the best way forward here is to simply keep on that path. If it turns out to be a major problem I suppose I'll have to release RepoBee 4 with a better idea for how to deal with it.

I think it should be sufficient here to just lowercase the name repo names as well, because it seems like that isn't being done at the moment. I'll have a look at it.

**Comment by slarse** [comment]
Having spent a couple of hours on this, the whole name normalization thing is a bit of a sticky pie to poke my fingers into. What I'll do instead of "fixing" this for everyone is to introduce an interop environment variable `REPOBEE_DISABLE_NAME_NORMALIZATION` that simply accepts all names as-is. This should fix your problem, without running the risk of ruining anything for others.

**Comment by DanielEWeeks** [comment]
Thank you for looking into it, and the proposed solution via an environmental variable sounds perfect.

And thank you also for earlier creating the 'ghclassroom' plugin that allows one to use repobee with the different repo naming scheme that is used in GitHub Classroom-created student repositories.

**Comment by slarse** [comment]
You're most welcome!

I've a branch at #1172 that I think should do the trick, but it took a bit too long to iron out some build issues so I won't be able to merge and release today. Feel free to try that branch out in the meantime, I hope to be able to get a new release out tomorrow.

**Comment by slarse** [comment]
Hi @DanielEWeeks,

I just released version 3.9.0 of RepoBee. If you set the environment variable install that version and set `REPOBEE_DISABLE_NAME_NORMALIZATION=true`, I think your problem should be resolved.
